### PR TITLE
talkdesk 2.0.1 (new cask)

### PR DESCRIPTION
**Pull request by DanzigerGeist**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

**NOTE**: Running `brew audit --cask --new <cask>` throws a warning: "_cask token mentions desktop_". I would like to request an exception since the official application name is Talkdesk Desktop.

### CODE REVIEW

**Comment by krehel** [comment]
> **NOTE**: Running `brew audit --cask --new <cask>` throws a warning: "_cask token mentions desktop_". I would like to request an exception since the official application name is Talkdesk Desktop.

It should be `talkdesk` because the app block is `Talkdesk.app`.  The DMG's in the Electron feed also refer to it as Talkdesk.

**Comment by DanzigerGeist** [comment]
@krehel thank you for taking a look on this one! I referred to the vendor's [documentation](https://support.talkdesk.com/hc/en-us/articles/13920616167963-Workspace-Desktop-App-for-Talkdesk-CX-Cloud) for the name, but having it just as "Talkdesk" should not cause any confusion - so I renamed it as per yours request.

File: Casks/t/talkdesk.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# https://support.talkdesk.com/hc/en-us/articles/13920616167963-Workspace-Desktop-App-for-Talkdesk-CX-Cloud
```
Comment (bevanjkay) [inline Casks/t/talkdesk.rb:1]:
```suggestion
```

File: Casks/t/talkdesk.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+cask "talkdesk" do
+  version "2.0.1-alpha.1"
+  sha256 "16eb1e4708ea74250231404caf18a472624dd67fc907c4f438d65fc0694c63c5"
+
+  url "https://td-infra-prd-us-east-1-s3-atlaselectron.s3.amazonaws.com/talkdesk-#{version}.dmg",
+      verified: "td-infra-prd-us-east-1-s3-atlaselectron.s3.amazonaws.com/"
+  name "Talkdesk Workspace Desktop for MacOS"
+  desc "Desktop app that allows you to use Talkdesk as stand-alone software"
+  homepage "https://www.mytalkdesk.com/atlas/download"
```
Comment (bevanjkay) [inline Casks/t/talkdesk.rb:10]: Is there a better url that can be used for the Homepage?

**Comment by krehel** [comment]
I'm not sure this is ready for inclusion quite yet.  Upstream appears to have some issues to work out.

I installed the version in this Cask, and it immediately force "updated" to version `1.12`, which does not seem to work in my VM.

So this needs to go back to `1.12` at least if the version here is considered unusable, and if `1.12` crashes then this should be tabled until some stability is in this app.

**Comment by krehel** [comment]
@DanzigerGeist - thanks for opening this, but I'm closing it for now as this version in the PR seems unstable, and the replacement version does not seem to work for me.  Please feel free to open this back up when and if this software stabilizes.

**Comment by DanzigerGeist** [comment]
@krehel I apologise for the delayed response - really tough week on the main job. I agree that the 2.0 acts somewhat weird, but I also noticed the electron builder returns the alpha version (unstable). Can we consider an option of switching the live check to use the versions listed [here](https://www.mytalkdesk.com/atlas/download) instead of the current approach, and set it to 1.12, which is indeed stable?

If you agree with the proposed approach - can we please have this one re-opened and I will work on it tomorrow.

**Pull request closed without merging**